### PR TITLE
584 soft client-side authentication

### LINKED ISSUE #584: Implement soft client-side authentication by only mounting dashboards if authorized is true
Example results from the API
```json
[{
	"name": "Program of Analysis",
	"version": "1.0",
	"owner": "MEME",
	"capability": "Utility",
	"runtime": "GO",
	"documentation": "https://edge.deciphernow.com/services/poa/1.0/",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": false,
	"instances": ["44b07f536aa315516103df29ebaa5b32"]
}, {
	"name": "Discovery Service",
	"version": "1.0",
	"owner": "Decipher",
	"capability": "Service Discovery",
	"runtime": "JVM",
	"documentation": "https://edge.deciphernow.com/services/discovery-service/1.0/home.html",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": true,
	"instances": ["1ea349346c473b5c50ef9d8899d73229"]
}, {
	"name": "Export Service",
	"version": "1.0",
	"owner": "MEME",
	"capability": "Utility",
	"runtime": "GO",
	"documentation": "https://edge.deciphernow.com/services/export-service/1.0/",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": false,
	"instances": ["24ffbb5db29b768717f7f2f6c2bb37ea"]
}, {
	"name": "Team Service",
	"version": "1.0",
	"owner": "MEME",
	"capability": "Organizations",
	"runtime": "GO",
	"documentation": "https://edge.deciphernow.com/services/team-service/1.0/",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": false,
	"instances": ["4b53fc697dac1764a16e3b93e6bc9cf2"]
}, {
	"name": "Up2 Service",
	"version": "1.0",
	"owner": "MEME",
	"capability": "Organizations",
	"runtime": "GO",
	"documentation": "https://edge.deciphernow.com/services/up2/1.0/",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": false,
	"instances": ["42d1a0651139eb7902acdd55f28bfdc0"]
}, {
	"name": "ICPF Service",
	"version": "1.0",
	"owner": "MEME",
	"capability": "Utility",
	"runtime": "GO",
	"documentation": "https://edge.deciphernow.com/services/icpf-service/1.0/",
	"minimum": 1,
	"maximum": 1,
	"authorized": true,
	"metered": true,
	"threaded": false,
	"instances": ["5692d2be9a99e5d199bb2b02abc6f05d"]
}]
```

### PR DESCRIPTION
**Pull request by kaitmore**
I ended up just checking whether a service was authorized in FabricRouter and conditionally rendering the InstanceRouter or GMServiceView based on that.

I used a UI kit notification to indicate a redirect just for now until we factor all of that out.

Resolves #584

### CODE REVIEW

File: src/components/fabric/FabricGrid.js [hunk 1]
```diff
@@ -35,7 +36,13 @@ class FabricGrid extends Component {
 
   componentDidMount() {
     // Refresh services from the Fabric Server every time this loads
+    const { location: { state } } = this.props;
     Actions.fetchServices();
+    if (state && !state.authorized)
```
Comment (spmcbride1201) [inline src/components/fabric/FabricGrid.js]: Please add some documentation explaining that this state is added by the router.

File: src/components/fabric/FabricGrid.js [hunk 2]
```diff
@@ -35,7 +36,13 @@ class FabricGrid extends Component {
...
+      notification("Authorization required", {
+        status: "danger",
+        timeout: 500
```
Comment (spmcbride1201) [inline src/components/fabric/FabricGrid.js]: This feels way too quick to me. I'd consider keeping it on the screen and forcing the user to dismiss it.

Comment (spmcbride1201) [reply src/components/fabric/FabricGrid.js]: The message would be a bit nicer if it was like "You are not authorized to view (insert microservice name here)". That would take some changes to your state.

**Comment by kaitmore** [comment]
@spmcbride1201 I edited the notification to include the service name and changed the cursor to 'not-allowed'. Also added comments to clarify where state is coming from in the `FabricGrid`

**Comment by kaitmore** [comment]
@spmcbride1201 links are now disabled for unauthorized services

File: src/components/Main/scenes/Fabric/components/FabricMainView/scene/Table/components/GMServiceList/components/GMServiceListItem.js [hunk 3]
```diff
@@ -48,6 +48,12 @@ const ItemVersion = styled.div`
   align-self: center;
 `;
 
+const ItemInfo = GMLink.withComponent("div").extend`
```
Comment (spmcbride1201) [inline src/components/Main/scenes/Fabric/components/FabricMainView/scene/Table/components/GMServiceList/components/GMServiceListItem.js]: Please refactor GMServiceListItem.js into a directory names GMServiceListItem with the logic in index.js and all of these components in a components subdirectory

**Comment by kaitmore** [comment]
@spmcbride1201 Refactored `GMServiceListItem` and `GMServiceCard` into new structure

**Pull request merged**